### PR TITLE
Fix slog handler edge cases

### PR DESCRIPTION
**Pull request by jan-law**
- logging nil pointer to `encoding.TextMarshaler`
- updates tests to reflect decision to enforce timestamps on all logs
- fixes typo for expected output when dropping an attribute

### CODE REVIEW

File: pkg/logger/color_handler.go [hunk 1]
```diff
@@ -335,6 +336,10 @@ func (h *colorHandler) appendValue(buf *buffer, v slog.Value, shouldQuote bool)
 		case slog.Level:
 			h.appendLevel(buf, cv)
 		case encoding.TextMarshaler:
+			if reflect.ValueOf(cv).IsNil() {
+				h.appendString(buf, "", shouldQuote)
+				break
+			}
```
Comment (kralicky) [inline pkg/logger/color_handler.go:342]: There's also line below here that it won't let me comment on
`h.appendString(buf, fmt.Sprint(v.Any()), shouldQuote)`
instead of calling v.Any() again can you use `cv`?
